roles/dns-records/tasks/main.yml

```diff
@@ -50,6 +50,22 @@
   with_items: "{{ groups['infra_hosts'] }}"
   when: hostvars[item]['public_v4'] is defined
 
+- name: "Add public master cluster hostname records to the public A records (single master)"
+  set_fact:
+    public_records: "{{ public_records | default([]) + [ { 'type': 'A', 'hostname': hostvars[groups.masters[0]].openshift_master_cluster_public_hostname, 'ip': hostvars[groups.masters[0]].public_v4 } ] }}"
```

Comment:
Change the `hostname` contents to:

    (hostvars[groups.masters[0]].openshift_master_cluster_public_hostname | replace(full_dns_domain, ''))[:-1]

That will turn say `api.openshift.example.com` to `api`, which is what we need for nsupdate.